packages/editor/src/components/post-featured-image/index.js

```diff
@@ -150,6 +155,25 @@ function PostFeaturedImage( {
 		);
 	}
 
+	const { isRequestingFeaturedImageMedia } = useSelect(
+		( select ) => {
+			const _isRequestingFeaturedImageMedia =
+				!! featuredImageId &&
+				! select( coreStore ).hasFinishedResolution( 'getMedia', [
+					featuredImageId,
+					{ context: 'view' },
+				] );
+
+			return {
+				isRequestingFeaturedImageMedia: _isRequestingFeaturedImageMedia,
+			};
+		},
+		[ featuredImageId ]
+	);
+
+	const noImageData =
+		! isRequestingFeaturedImageMedia && !! featuredImageId && ! media;
```

Comment:
Nitpick: Using `isMissingMedia` or something similar might be more descriptive here.